### PR TITLE
Fix container timezone

### PR DESCRIPTION
**Pull request by gjrtimmer**
***Gitlab can have a different timezone in relation to the container***

# Introduction
Default the containers run with timezone set to `UTC`, which means that the cron daemon can be using a different timezone than gitlab, because the timezone from gitlab can be configured with `GITLAB_TIMEZONE`.

# Use case
My current TimeZone is `Europe/Amsterdam` Gitlab has the corrent time, but because the container time is `UTC` it means that the cron daemon which runs the backups does this ` hour to early.

# Provided solution

* ADDED 
  * `configure_container_timezone()` -> `assets/runtime/functions`
* MODIFY
  * Added `configure_container_timezone()` to function `configure_system()` before the function `install_configuration_templates` because timezone is a lower level then templates.
  * README
    * ADD: Variable `TZ`
    * UPD: Variable `GITLAB_TIMZONE`

## configure_container_timezone()

The provided function has a sanity check which will use `UTC` as default. The sanity check will check that the user provided value is a valid timezone, by checking if the provided timezone exists within the `tzdata` zoneinfo.

# Configuration

Container TimeZone will use the environment variable `TZ`, and defaults to `UTC`. The variable `TZ` expects values in Canonical format. Link to acceptable values has been added to the  README. The format which is expected causes that the `GITLAB_TIMEZONE` cannot be used here because `GITLAB_TIMEZONE` can accept timezone in different formats.

### CODE REVIEW

**Comment by GJRTimmer** [comment]
@solidnerd Request review

**Comment by splattael** [comment]
@GJRTimmer Thanks for the PR!

Would mounting `/etc/localtime` into the container have the same effect? E.g. `-v /etc/localtime:/etc/localtime:ro`?

**Comment by GJRTimmer** [comment]
@splattael Never tested that, and I don't know if it would be a solution. I think there would be quite a few situations where there is a complete different timezone between the containers and the host. With this fix it the container will get the correct timezone. Got this fix from `docker` on how to set the container timezone correctly by using the `TZ` variable.

This image might even be used by parties that are hosting gitlab environments for clients in different timezone. So I don't want to mount the localtime from the host I want to set it correctly.

**Comment by GJRTimmer** [comment]
@solidnerd @sameersbn Any Update

**Comment by GJRTimmer** [comment]
Rebased on new master

**Comment by GJRTimmer** [comment]
@solidnerd Rebased on new master

**Comment by GJRTimmer** [comment]
@solidnerd Rebased to 11.5.0

**Comment by GJRTimmer** [comment]
Rebased to 11.5.1

**Comment by GJRTimmer** [comment]
@solidnerd rebased

**Comment by GJRTimmer** [comment]
@solidnerd Rebased

**Comment by drubin** [comment]
@GJRTimmer Hi is there a reason you don't just set the `TZ` env variable?

File: assets/runtime/functions [hunk 1]
```diff
@@ -1448,12 +1448,30 @@ generate_ssh_host_keys() {
   chmod 0644 ${GITLAB_DATA_DIR}/ssh/*.pub
 }
 
+configure_container_timezone() {
+  TIMEZONE=${TZ:-UTC}
```
Comment (drubin) [inline assets/runtime/functions]: This default config I think should be set in `env-defaults` if it is in fact required.

Comment (sameersbn) [reply assets/runtime/functions:1452]: +1

**Comment by sameersbn** [comment]
sorry about the delay in attending to this PR. I really have no issues getting it merged but I am just curious to know if there is a use-case that just setting the `TZ` env variable is not suffice?

**Comment by GJRTimmer** [comment]
> sorry about the delay in attending to this PR. I really have no issues getting it merged but I am just curious to know if there is a use-case that just setting the `TZ` env variable is not suffice?

Sorry for the delay; I have environment variable TZ=Europe/Amsterdam set on the container. Still the shell is not reporting the current correct time which results that for example my backup files have an offset of 2 hours because of GMT+2 (Daylight Saving Time).

Proposed PR would fix that.

File: assets/runtime/functions (hunk 1 above)
Comment (GJRTimmer) [reply assets/runtime/functions]: PR Updated;

**Comment by GJRTimmer** [comment]
@solidnerd @sameersbn Rebased & Implemented conversation suggestion

**Comment by GJRTimmer** [comment]
@solidnerd @sameersbn Why is the build not completing ?

**Comment by sameersbn** [comment]
needs a rebase.

**Comment by gjrtimmer** [comment]
@sameersbn Rebased, very sorry for the delay

**Review by sameersbn — APPROVED** [review]

**Pull request merged**